### PR TITLE
Allow birdseye to be overridden at the camera level

### PR DESCRIPTION
**Pull request by NickM-27**
Implementation of #2582

### CODE REVIEW

File: frigate/output.py [hunk 1]
```diff
@@ -310,11 +315,14 @@ def update_frame(self):
...
-    def update(self, camera, object_count, motion_count, frame_time, frame) -> bool:
+    def update(self, camera, camera_config, object_count, motion_count, frame_time, frame) -> bool:
+        # don't process if birdseye is disabled for this camera
+        if camera_config and not camera_config.enabled:
+            return False
 
         # update the last active frame for the camera
         self.cameras[camera]["current_frame"] = frame_time
-        if self.camera_active(object_count, motion_count):
+        if self.camera_active(camera_config, object_count, motion_count):
```
Comment (blakeblackshear) [inline frigate/output.py:325]: Just pass `camera_config.mode`. The config parser should ensure that that value is always set.

File: frigate/output.py [hunk 2]
```diff
@@ -310,11 +315,14 @@ def update_frame(self):
 
         return True
 
-    def update(self, camera, object_count, motion_count, frame_time, frame) -> bool:
+    def update(self, camera, camera_config, object_count, motion_count, frame_time, frame) -> bool:
+        # don't process if birdseye is disabled for this camera
+        if camera_config and not camera_config.enabled:
```
Comment (blakeblackshear) [inline frigate/output.py:320]: The config parser should make it impossible for `camera_config` to be None here.

```python
if not camera_config.enabled:
```

File: frigate/config.py [hunk 3]
```diff
@@ -775,6 +785,7 @@ def runtime_config(self) -> FrigateConfig:
         # Global config to propegate down to camera level
         global_config = config.dict(
             include={
+                "birdseye": ...,
```
Comment (blakeblackshear) [inline frigate/config.py:788]: If you disable globally and then enable a single camera, does that enable Birdseye for the one camera?

**Review by blakeblackshear — CHANGES_REQUESTED** [review]

File: frigate/config.py (hunk 3 above)
Comment (NickM-27) [reply frigate/config.py:788]: With birdseye disabled globally, there is no option in the sidebar. If I bypass that then it still isn't enabled as the global check to add the converter["birdseye"] doesn't go through. Not sure what the best behavior would be; I could add a warning in the config if it is disabled globally, or adapt those behaviors to consider the camera level as well, or perhaps it is good as is.

**Comment by NickM-27** [comment]
> Might be a good idea to add some unit tests for the additional config parsing to make sure it handles things the way we expect.

Yeah good idea, I think the two I added should cover the cases but let me know if I missed any 👍

**Pull request merged**